src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java

```diff
@@ -78,9 +78,6 @@
     private transient @Deprecated Integer startTimeout;
     private transient @Deprecated Boolean floatingIps;
 
-    // Cache the instance between uses
-    private volatile transient @CheckForNull Openstack openstack;
```

Comment:
My intention here was to ensure that threads that repeatedly talk to the same cloud at about the same time (AJAX calls for form validation) would reuse the same instance of {{Openstack}} that will take care of creating dedicated client instance based on existing token. IOW, one auth dance per instance. This has the obvious flaw of not expiring the instance.

As I understand the change, the logic was moved but the behavior is the same plus it invalidates with fixed time.

Reply:
Correct - the (old) functionality in JCloudsCloud meant that the Openstack instance was never expired.
This PR keeps the same same error handling but now every call to `JCloudsCloud.getOpenstack()` delegates to `Openstack.Factory.get(...)` and Openstack.Factory provides both caching _and_ token expiry.

i.e. the old code did caching without expiry, the new code provides caching with expiry.